src/txorphanage.h

```diff
@@ -22,7 +22,7 @@ class TxOrphanage {
 public:
     /** Add a new orphan transaction. If the tx already exists, add this peer to its list of announcers.
       @returns true if the transaction was added as a new orphan. */
-    bool AddTx(const CTransactionRef& tx, NodeId peer) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex);
+    bool AddTx(const CTransactionRef& tx, NodeId peer, const std::vector<uint256>& parent_txids) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex);
```

Comment:
Should describe the new argument as something like "all unique parent transaction txids"